numpy/core/numerictypes.py

```diff
@@ -523,7 +523,7 @@ def sctype2char(sctype):
     ScalarType = [int, float, complex, int, bool, bytes, str, memoryview]
 
 ScalarType.extend(_concrete_types)
-ScalarType = tuple(ScalarType)
+ScalarType = frozenset(ScalarType)
```

Comment:
I'm not sure we should change this, someone may be relying on the order.

Reply:
Agreed.